notmuch/mutt_notmuch.c

```diff
@@ -1060,18 +1110,23 @@ static bool read_mesgs_query(struct Mailbox *m, notmuch_query_t *q, bool dedup)
   if (!msgs)
     return false;
 
+  header_cache_t *h = nm_hcache_open(m);
```

Comment:
[1] ... particularly in places like here.

Imagine if I compiled Neomutt w/o hcache and decided to debug its behavior. If I see a function call for hcache, it will lead to some confusion. I explicitly compiled without a header cache so there should be no traces of it.

Does this seem like an actual issue or am I being too pedantic?

Reply:
> I explicitly compiled without a header cache so there should be no traces of it.

Agreed.

We need some standardisation between the backends on _how_ they use the header cache.

> Does this seem like an actual issue or am I being too pedantic?

It's a mess, but I think we should commit the changes...

and start thinking about how caching should work in a perfect situation.